processor.go

```diff
@@ -624,15 +623,26 @@ func (p *CertProcessor) processCertificate(cert Certificate, forMaint bool) (boo
 		return p.NoteCertError(cert, err, "Error while marshalling altNames for domain %v", cert.Spec.Domain)
 	}
 
-	// Save cert details to bolt
-	err = p.db.Update(func(tx *bolt.Tx) error {
-		key := []byte(cert.Spec.Domain)
-		tx.Bucket([]byte("cert-details")).Put(key, certDetailsRaw)
-		tx.Bucket([]byte("domain-altnames")).Put(key, altNamesRaw)
-		return nil
-	})
+	//Need to distinguish if this is a new cert or a renewal.
+	if isRenewal {
+		err = updateCertDetails(cert.Spec.Domain, certDetailsRaw, p.db)
+	} else {
+		err = addCertDetails(cert.Spec.Domain, certDetailsRaw, p.db)
+	}
```

Comment:
This is going to be nitpicky, but would it be slightly cleaner to have a `saveCertDetails` function where you pass in `isRenewal` and do the logic to add vs update there?

Reply:
It would look something like 
`func saveCertDetails(domain string, certDetailsRaw []byte, d *gorm.DB, renewal bool) (err error) {
	s := string(certDetailsRaw)
	if renewal {
		var certDetails CertDetail
		if dbErr := d.Where(&CertDetail{Domain: domain}).First(&certDetails); dbErr != nil {
			return errors.Wrapf(err, "GORM error %s", dbErr)
		}
		certDetails.Value = s
		d.Save(&certDetails)
	} else {
		if dbErr := d.Create(&CertDetail{Domain: domain, Value: s}); dbErr != nil {
			return errors.Wrapf(err, "GORM error %s", dbErr)
		}
	}
	return err
}`

Reply:
Updated db.go and added a save function.